syntax = "proto3";

package caption_service.v1;

import "google/api/annotations.proto";
import "messages/v1/image_dto.proto";

// Request to get caption for an image
message GetCaptionRequest {
	shared.messages.v1.ImageDto image = 1;
}

// Response containing caption for an image
message GetCaptionResponse {
    string caption = 1;
}

// Service to handle all image captioning operations
service CaptionService {
    // Get caption for an image
	rpc get_caption(GetCaptionRequest) returns (GetCaptionResponse) {
		option (google.api.http) = {
			post: "/v1/caption"
			body: "*"
		};
	}
}
